app/CMakeLists.txt

```diff
@@ -288,6 +288,88 @@ else()
       "${app_desktop_HDRS}")
 endif()
 
+if(IOS)
+  file(GLOB admob_HDRS
+    "${FIREBASE_SOURCE_DIR}/admob/src/include/firebase/admob.h"
+    "${FIREBASE_SOURCE_DIR}/admob/src/include/firebase/admob/*.h")
+
+  # Generate the analytics header file
+  # Copy from analytics/CMakeList.txt
+  function(generate_analytics_header OBJC_FILE CPP_FILE)
+    add_custom_command(
+      OUTPUT ${CPP_FILE}
+      COMMAND python "${FIREBASE_SOURCE_DIR}/analytics/generate_constants.py"
+        "--objc_header=${OBJC_FILE}"
+        "--cpp_header=${CPP_FILE}"
+      DEPENDS ${OBJC_FILE}
+      COMMENT "Generating ${CPP_FILE}"
+    )
+  endfunction()
+  set(analytics_generated_headers_dir
+    "${FIREBASE_GEN_FILE_DIR}/analytics/src/include/firebase/analytics")
+  file(MAKE_DIRECTORY "${FIREBASE_GEN_FILE_DIR}/analytics/src/include/firebase/analytics")
+  generate_analytics_header(
+    "${FIREBASE_SOURCE_DIR}/analytics/ios_headers/FIREventNames.h"
+    "${analytics_generated_headers_dir}/event_names.h"
+  )
+  generate_analytics_header(
+    "${FIREBASE_SOURCE_DIR}/analytics/ios_headers/FIRParameterNames.h"
+    "${analytics_generated_headers_dir}/parameter_names.h"
+  )
+  generate_analytics_header(
+    "${FIREBASE_SOURCE_DIR}/analytics/ios_headers/FIRUserPropertyNames.h"
+    "${analytics_generated_headers_dir}/user_property_names.h"
+  )
+  set(analytics_HDRS
+    "${FIREBASE_SOURCE_DIR}/analytics/src/include/firebase/analytics.h"
+    "${analytics_generated_headers_dir}/event_names.h"
+    "${analytics_generated_headers_dir}/parameter_names.h"
+    "${analytics_generated_headers_dir}/user_property_names.h")
+
+  file(GLOB auth_HDRS
```

Comment:
Though GLOB makes it easy to read and follow the CMake logic, it can have surprising side effects. If an explicit list is not crazy long, I prefer explicitly listing down the files.

https://stackoverflow.com/questions/32411963/why-is-cmake-file-glob-evil/32412044
Example from CMakeLists in our cpp sdk repo,
https://github.com/firebase/firebase-cpp-sdk/blob/master/database/CMakeLists.txt#L97

Not a big deal if you want to keep it the same as it is now.